tests/unit/checkpoint/test_moe.py

```diff
@@ -0,0 +1,119 @@
+from deepspeed.moe.utils import split_params_into_different_moe_groups_for_optimizer
+
+from tests.unit.common import DistributedTest
+from tests.unit.simple_model import *
+from tests.unit.util import required_torch_version
+
+from tests.unit.checkpoint.common import checkpoint_correctness_verification
+
+import pytest
+
+
+class TestMoECheckpoint(DistributedTest):
+    world_size = 4
+
+    @pytest.mark.parametrize("ep_size", [4])
+    def test_checkpoint_moe(self, tmpdir, ep_size):
+        if not required_torch_version():
+            pytest.skip("DeepSpeed MoE tests need torch 1.8 or higher to run correctly")
+
+        config_dict = {
+            "train_batch_size": 8,
+            "steps_per_print": 1,
+            "fp16": {
+                "enabled": True
+            }
+        }
+        hidden_dim = 16
+        args = args_from_dict(tmpdir, config_dict)
+
+        def _helper(args):
+            models = [
+                SimpleMoEModel(hidden_dim=hidden_dim,
+                               num_experts=ep_size,
+                               ep_size=ep_size) for _ in range(2)
+            ]
+            optimizers = [
+                torch.optim.AdamW(params=model.parameters()) for model in models
+            ]
+            checkpoint_correctness_verification(args,
+                                                models=models,
+                                                hidden_dim=hidden_dim,
+                                                tmpdir=tmpdir,
+                                                load_optimizer_states=True,
+                                                load_lr_scheduler_states=False,
+                                                fp16=config_dict["fp16"]["enabled"],
+                                                empty_tag=True,
+                                                base_optimizers=optimizers,
+                                                seq_dataloader=True)
```

Comment:
What's the purpose of having the local function `_helper`? couldn't this just be un-indented once and run the same? I think this applies to multiple tests here

Reply:
Fixed.